docs/scos/dev/glue-api-guides/202311.0/migrating-old-resource-to-new-infrastructure.md

```diff
@@ -0,0 +1,263 @@
+
+        $availabilityNotificationSubscriptionTransfer = (new AvailabilityNotificationSubscriptionTransfer())
+            ->fromArray($restAvailabilityNotificationRequestAttributesTransfer->toArray(), true)
+            ->setLocale($localeTransfer)
+            ->setStore($storeTransfer)
+            ->setCustomerReference($customerReference);
+
+        $availabilityNotificationSubscriptionResponseTransfer = $this->availabilityNotificationClient->subscribe($availabilityNotificationSubscriptionTransfer);
+
+        //Response mapped to the GlueResponseTransfer. For the sake of visibility code will be in this class.
+        if (!$availabilityNotificationSubscriptionResponseTransfer->getIsSuccess()) {
+            return $this->createSubscribeErrorResponse($availabilityNotificationSubscriptionResponseTransfer);
+        }
+
+        return $this->createSubscribeResponse($availabilityNotificationSubscriptionResponseTransfer);
+    }
+
+    protected function createSubscribeErrorResponse(AvailabilityNotificationSubscriptionResponseTransfer $availabilityNotificationSubscriptionResponseTransfer): GlueResponseTransfer
+    {
+        $errorPayload = $this->config->getErrorIdentifierToRestErrorMapping()[$availabilityNotificationSubscriptionResponseTransfer->getErrorMessage()] ?? $this->config->getDefaultSubscribeRestError();
+
+        return (new GlueResponseTransfer())
+            ->setHttpStatus($errorPayload[RestErrorMessageTransfer::STATUS])
+            ->addError(
+                (new GlueErrorTransfer())
+                    ->setStatus($errorPayload[RestErrorMessageTransfer::STATUS])
+                    ->setMessage($errorPayload[RestErrorMessageTransfer::DETAIL])
+                    ->setCode($errorPayload[RestErrorMessageTransfer::CODE]),
+            );
+    }
+
+    protected function createSubscribeResponse(AvailabilityNotificationSubscriptionResponseTransfer $availabilityNotificationSubscriptionResponseTransfer): GlueResponseTransfer
+    {
+        $restAvailabilityNotificationsAttributesTransfer = $this
+            ->availabilityNotificationMapper
+            ->mapAvailabilityNotificationSubscriptionTransferToRestAvailabilityNotificationsAttributesTransfer(
+                $availabilityNotificationSubscriptionResponseTransfer->getAvailabilityNotificationSubscriptionOrFail(),
+                new RestAvailabilityNotificationsAttributesTransfer(),
+            );
+        
+        return (new GlueResponseTransfer())
+            ->setHttpStatus(Response::HTTP_OK)
+            ->addResource((new GlueResourceTransfer())
+                ->setType(OauthApiConfig::RESOURCE_TOKEN)
+                ->setAttributes($restAvailabilityNotificationsAttributesTransfer)
+            );
+    }
+}
+
+```
+
+We changed usage of RestResponse and RestRequest with usage of provided transfers. It will require some additional steps and not always easy like in provided example, so we can do it in a different way.
```

Comment:
"It will require some additional steps and not always easy like in provided example, so we can do it in a different way." Okay? so what does this mean?